typing/env.ml

```diff
@@ -1636,7 +1636,16 @@ and check_value_name name loc =
   (* Note: we could also check here general validity of the
      identifier, to protect against bad identifiers forged by -pp or
      -ppx preprocessors. *)
-  if String.length name > 0 && (name.[0] = '#') then
+  let is_identchar c =
+    (* identchar class in the lexer *)
```

Comment:
Could you insert an explicit comment, both here and in `lexer.mll` to state that the two definitions must be kept in sync?

Or get rid of this test and check for `#` in all identifiers? Regular identifiers are not supposed to contain this character anyway, right?

Reply:
Comments added here and in `lexer.mll`.